Lombiq.Tests.UI/SecurityScanning/ZapManager.cs

```diff
@@ -0,0 +1,244 @@
+#pragma warning restore S103 // Lines should not be too long
+
+        var cliParameters = new List<object> { "run" };
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        {
+            cliParameters.Add("--network");
+            cliParameters.Add("host");
+        }
+        else
+        {
+            cliParameters.Add("--add-host");
+            cliParameters.Add("localhost:host-gateway");
+        }
+
+        cliParameters.AddRange(new object[]
+        {
+            "--volume",
+            $"{mountedDirectoryPath}:{_zapWorkingDirectoryPath}:rw",
+            "--tty",
+            _zapImage,
+            "zap.sh",
+            "-cmd",
+            "-autorun",
+            _zapWorkingDirectoryPath + yamlFileName,
+        });
+
+        var stdErrBuffer = new StringBuilder();
+
+        // Here we use a new container instance for every scan. This is viable and is not an overhead big enough to
+        // worry about, but an optimization would be to run multiple scans (also possible simultaneously with background
+        // commands) with the same instance. This needs the -dir option to configure a different home directory per
+        // scan, see https://www.zaproxy.org/docs/desktop/cmdline/#options.
+        var result = await _docker
+            .GetCommand(cliParameters)
+            .WithStandardOutputPipe(PipeTarget.ToDelegate(line => _testOutputHelper.WriteLineTimestampedAndDebug(line)))
+            .WithStandardErrorPipe(PipeTarget.ToStringBuilder(stdErrBuffer))
+            // This is so no exception is thrown by CliWrap if the exit code is not 0.
+            .WithValidation(CommandResultValidation.None)
+            .ExecuteAsync(_cancellationTokenSource.Token);
+
+        _testOutputHelper.WriteLineTimestampedAndDebug("Security scanning completed with the exit code {0}.", result.ExitCode);
+
+        if (result.ExitCode == 1)
+        {
+            throw new SecurityScanningException("Security scanning didn't successfully finish. Check the test's output log for details.");
+        }
+
+        var jsonReports = Directory.EnumerateFiles(reportsDirectoryPath, "*.json").ToList();
+
+        if (jsonReports.Count > 1)
+        {
+            throw new SecurityScanningException(
+                "There were more than one JSON reports generated for the ZAP scan. The supplied ZAP Automation " +
+                "Framework YAML file should contain exactly one JSON report job, generating a SARIF report.");
+        }
+
+        if (jsonReports.Count != 1)
+        {
+            throw new SecurityScanningException(
+                "No SARIF JSON report was generated for the ZAP scan. This indicates that the scan couldn't finish. " +
+                "Check the test output for details.");
+        }
+
+        return new SecurityScanResult(reportsDirectoryPath, SarifLog.Load(jsonReports[0]));
```

Comment:
Not sure if this is the right spot, but somewhere around here the used image should also be deleted (not a problem for an ephemeral CI runner, but interesting in local dev environment for example.).

Reply:
Yeah, and also the container.

Reply:
Sorry, I meant that we should delete the containers, but IMO keeping the image is OK, so you don't have to download it for every run (it's 2GB).